kong/plugins/prometheus/exporter.lua

```diff
@@ -258,10 +275,34 @@ else
       route_name = message.route.name or message.route.id
     end
 
+    local consumer = ""
+    if message and serialized.consumer ~= nil then
+      consumer = serialized.consumer
+    end
+
+
     labels_table[1] = service_name
     labels_table[2] = route_name
     labels_table[3] = message.session.status
-    metrics.status:inc(1, labels_table)
+    labels_table[4] = consumer
+
+    latency_labels_table[1] = service_name
+    latency_labels_table[2] = route_name
+
+    labels_table_status[1] = service_name
+    labels_table_status[2] = route_name
+    labels_table_status[3] = message.session.status
+    labels_table_status[5] = consumer
+
+    -- If number of upstream requests is greater than 0, assume upstream is the status code originator
+    -- Else, assume kong is the originator.
+    if(#(message.tries) > 0) then
```

Comment:
Is this assumption correct? What if the balancer tries up to a `max`, gives up and then Kong responds back with a 5xx?
@locao @fffonion ?

Reply:
I understand the desire to be 100% correct, but I think in a system with arbitrary plugin code execution this is hard to get 100% correct.

Do you think this is good enough for a first pass? 

@sichvoge do you have thoughts on the matter?

Reply:
Please confirm this with @locao before merging.

Reply:
No particular thoughts to be honest since I am not knowledgeable enough about all the different cases. Just to be sure, we should double check with @locao as Harry suggested.

Reply:
we can use `kong.response.get_source()` instead